### PR TITLE
Feature/32 create endpoint to get all polls with votes

### LINKED ISSUE #32: Create endpoint to get all polls with votes
Create endpoint `polls/all`. The endpoint should return a json with all polls and the calculated votes.

### PR DESCRIPTION
**Pull request by GrafBunsen**
closing #32

### CODE REVIEW

File: app/Http/Controllers/PollController.php [hunk 1]
```diff
@@ -34,4 +35,31 @@ public function create(Request $request): JsonResponse
...
+                "phase" => $poll->phase,
+                "created_at" => $poll->created_at,
+                "updated_at" => $poll->updated_at,
+                "upvotes" => $upvotes,
+                "downvotes" => $downvotes,
+            ];
+
+            $convertedPolls[] = $convertedPoll;
+        }
+        return response()->json($convertedPolls);
```
Comment (martincodes-de) [inline app/Http/Controllers/PollController.php:65]: please before return, add empty line

**Review by martincodes-de — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/PollController.php [hunk 2]
```diff
@@ -34,4 +35,31 @@ public function create(Request $request): JsonResponse
...
+        foreach ($polls as $poll) {
+            $votes = Vote::where('poll_id', $poll->id)->get();
+            $upvotes = $votes->where('description', true)->count();
+            $downvotes = $votes->where('description', false)->count();
+
+            $convertedPoll = [
+                "id" => $poll->id,
+                "name" => $poll->name,
+                "description" => $poll->description,
+                "endDate" => $poll->end_date,
```
Comment (martincodes-de) [inline app/Http/Controllers/PollController.php:54]: to be consistent here as api, i would recommend snake case like `updated_at` and so on.

File: app/Http/Controllers/PollController.php [hunk 3]
```diff
@@ -34,4 +35,31 @@ public function create(Request $request): JsonResponse
...
+                "phase" => $poll->phase,
+                "created_at" => $poll->created_at,
+                "updated_at" => $poll->updated_at,
+                "upvotes" => $upvotes,
+                "downvotes" => $downvotes,
```
Comment (martincodes-de) [inline app/Http/Controllers/PollController.php:57]: would move up and downvotes before created_at and updated_at

**Review by martincodes-de — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/PollController.php (hunk 1 above)
Comment (GrafBunsen) [reply app/Http/Controllers/PollController.php:65]: done

File: app/Http/Controllers/PollController.php (hunk 2 above)
Comment (GrafBunsen) [reply app/Http/Controllers/PollController.php:54]: we use lower camel case in the rest of the API, don't we? so change update-at to updateAt?

File: app/Http/Controllers/PollController.php (hunk 3 above)
Comment (GrafBunsen) [reply app/Http/Controllers/PollController.php:57]: done

**Review by martincodes-de — DISMISSED** [review]

File: app/Http/Controllers/PollController.php (hunk 2 above)
Comment (GrafBunsen) [reply app/Http/Controllers/PollController.php:54]: changed to camel case

File: app/Http/Controllers/PollController.php [hunk 4]
```diff
@@ -32,6 +33,35 @@ public function create(Request $request): JsonResponse
...
         return response()->json($poll, status: 201);
     }
+
+    public function getAll(): JsonResponse
+    {
+        $polls = Poll::all();
+        $convertedPolls = [];
+
+        foreach ($polls as $poll) {
+            $votes = Vote::where('poll_id', $poll->id)->get();
```
Comment (martincodes-de) [inline app/Http/Controllers/PollController.php]: Please use $votes = $poll->votes() instead of a new query

**Review by martincodes-de — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/PollController.php (hunk 4 above)
Comment (GrafBunsen) [reply app/Http/Controllers/PollController.php]: done

**Review by martincodes-de — APPROVED** [review]
Now lgtm

**Pull request merged**